packages/smooth_app/lib/pages/product/add_new_product_page.dart

```diff
@@ -77,22 +77,34 @@ class _AddNewProductPageState extends State<AddNewProductPage> {
           automaticallyImplyLeading: !_isProductLoaded),
       floatingActionButton: FloatingActionButton.extended(
         onPressed: () async {
-          final LocalDatabase localDatabase = context.read<LocalDatabase>();
-          final DaoProduct daoProduct = DaoProduct(localDatabase);
-          final Product? localProduct = await daoProduct.get(widget.barcode);
-          if (localProduct == null) {
-            product = Product(
-              barcode: widget.barcode,
-            );
-            daoProduct.put(product);
-            localDatabase.notifyListeners();
-          }
-          localDatabase.upToDate.set(product);
-          if (mounted) {
-            await Navigator.maybePop(
-              context,
-              _isProductLoaded ? widget.barcode : null,
-            );
+          if (_isProductLoaded ||
```

Comment:
Just for the record, it's probably a best practice in tests to get rid of the simplest cases first and to return.
Instead of jumping 20 lines of code with a simple else ("btw else of what?").
That's the way I code for ages (that doesn't mean it's a best practice) and I even found a video on youtube about it a couple of weeks ago (that doesn't mean it's a best practice either).

Reply:
Well initially I thought that first but if we see practically most don't return without adding products from the add page, 

So can say safely that , here putting it later was perhaps a good idea. Your call, should I fix it or leave it as it is?

Reply:
It's not a matter of what the user is expected to do, it's a matter of escaping the code ASAP.
I think it's easier to start a method with `if(param == null || param.isEmpty) { return null;}` and then do what is interesting, without curly braces and a pending `else`.
On the other hand your `if` test is easy to understand (you need at least one field populated, right), and it would look like crap with a logical NOT on top.
Let's keep it the way it is now.